server/src/main/java/org/apache/druid/indexing/overlord/supervisor/Supervisor.java

```diff
@@ -93,6 +94,14 @@ default Boolean isHealthy()
    */
   LagStats computeLagStats();
 
+  /**
+   * Used by AutoScaler to either scale by max/total/avg.
+   */
+  default LagMetric getLagMetricForAutoScaler()
```

Comment:
This is a very roundabout and yet limiting manner of determining which lag metric to use for auto-scaling.

We don't need the `LagMetric` class or the `Supervisor.getLagMetric` method.

We can simply add another method `computeLagForAutoScaler()`. This clarifies the intent and is not confusing.

```java
default long computeLagForAutoScaler() {
    LagStats lagStats = computeLagStats();
    return lagStats == null ? 0L : lagStats.maxLag();
}
```

Update the `LagBasedAutoScaler` to use this new method instead of `computeLagStats()`.

cc: @adithyachakilam , @AmatyaAvadhanula